### ISSUE TITLE
[Feature Request] first render performance improvement with defer umi.js

### ISSUE THREAD

**Issue by KrishnaPG** [umijs/umi#7885]
## Background
Currently the `umi.js` is loaded in a blocking way in the page. This is very bad for performance. Check the below screenshot:

![image](https://user-images.githubusercontent.com/4610831/155829509-912e4619-f8d6-467f-90df-8fdc35208832.png)

Because `umi.js` is not deferred, all the above deferred scripts fail to give correct loading results.
## Proposal

use defer for the umi.js as shown below. It drastically improves performance.
![image](https://user-images.githubusercontent.com/4610831/155829665-15193b01-c25e-4163-b086-43389f6ac529.png)

## Additional context

Using `defer` is a safe and good practice for modern websites that improve the first-paint performance. For those of you who has doubts about it, or do not understand what it is, please refer the below articles:

- `Scripts with the defer attribute will execute in the order in which they appear in the document.` https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script#attr-defer
- `Adding defer to both your vendor script and your main script will cause them to **load in parallel**, not block parsing, but **processed in order** on document parse completion.` https://stackoverflow.com/questions/42615758/javascript-async-defer-order-of-execution-in-case-of-multiple-files-in-single-pa
- **Modules are deferred by default** https://v8.dev/features/modules#defer
![image](https://user-images.githubusercontent.com/4610831/155829860-45b71215-19b3-4049-93d2-e7b0f96dcdad.png)
